corehq/ex-submodules/couchforms/dbaccessors.py

```diff
@@ -88,16 +90,43 @@ def get_form_ids_for_user(domain, user_id):
     return [result['id'] for result in results]
 
 
-def get_form_ids_by_xmlns(domain, xmlns=None):
+def iter_form_ids_by_xmlns(domain, xmlns):
     if xmlns:
         key = ['submission xmlns', domain, xmlns]
     else:
-        key = ['submission xmlns', domain]
+        key = ['submission', domain]
+
+    endkey = key + [datetime.utcnow().isoformat()]
+
+    # pull the first 1000 documents sorted by submission time
+    LIMIT = 1000
     results = XFormInstance.get_db().view(
         'all_forms/view',
         startkey=key,
-        endkey=key + [{}],
+        endkey=endkey,
         reduce=False,
         include_docs=False,
-    )
-    return [result['id'] for result in results]
+        limit=LIMIT,
+        descending=False,
+    ).all()
+
+    while results:
+        for result in results:
+            yield result['id']
+
+        # add the last document's submitted_on time to the startkey
```

Comment:
this naming is confusing since it's actually the received on time which is in the view key
```suggestion
        # add the last document's received_on time to the startkey
```